core/src/main/scala/kafka/server/AddPartitionsToTxnManager.scala

```diff
@@ -80,10 +81,9 @@ class AddPartitionsToTxnManager(
     topicPartitions: Seq[TopicPartition],
     callback: AddPartitionsToTxnManager.AppendCallback
   ): Unit = {
-    val (error, node) = getTransactionCoordinator(partitionFor(transactionalId))
-
-    if (error != Errors.NONE) {
-      callback(topicPartitions.map(tp => tp -> error).toMap)
+    val coordinatorNode = getTransactionCoordinator(partitionFor(transactionalId))
+    if (coordinatorNode.isEmpty) {
+      callback(topicPartitions.map(tp => tp -> Errors.COORDINATOR_NOT_AVAILABLE).toMap)
```

Comment:
we only returned either of (COORDINATOR_NOT_AVAILABLE, NoNode) or (NONE, SomeNode). Returning an option and translating to the appropriate error in the caller seemed simpler.